WireNetwork/Sources/WireNetwork/Network/NetworkStack/REST API/AuthenticatedRESTAPI.swift

```diff
@@ -0,0 +1,90 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+public struct AuthenticatedRESTAPI {
```

Comment:
This is a factory for all api clients that need an `APIService` (which requires an `AuthenticationManager`). It's an `async throws` operation to create this factory because the api version needs to be resolved.

But once we have it we can create api clients when needed.